packages/synapse-react-client/src/components/EntityUpload/LinkToURL.tsx

```diff
@@ -0,0 +1,93 @@
+import { Alert, Box, Stack, TextField, Typography } from '@mui/material'
+import { FileEntity } from '@sage-bionetworks/synapse-types'
+import { noop } from 'lodash-es'
+import React, { useEffect, useImperativeHandle, useState } from 'react'
+import { useGetEntity } from '../../synapse-queries/index'
+import { useLinkFileEntityToURL } from '../../utils/hooks/useUploadFileEntity/useLinkFileEntityToURL'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+
+type LinkToURLProps = {
+  /** The ID of the FileEntity that should be updated, or container entity in which a new FileEntity should be created. */
+  entityId: string
+  /** Invoked when the validity of the form changes (e.g. to disable/activate a submit button) */
+  onIsValidChanged?: (isValid: boolean) => void
+  /** Invoked when the call to create or update the entity succeeds. */
+  onSuccess?: (newOrUpdatedEntity: FileEntity) => void
+}
+
+export type LinkToURLHandle = {
+  submit: () => void
+}
+
+export const LinkToURL = React.forwardRef(function LinkToURL(
+  props: LinkToURLProps,
+  ref: React.ForwardedRef<LinkToURLHandle>,
+) {
+  const { entityId, onIsValidChanged = noop, onSuccess = noop } = props
+
+  const [url, setUrl] = useState('')
+  const [name, setName] = useState('')
+
+  const { data: entity } = useGetEntity(entityId)
+  const isFileEntity =
+    entity?.concreteType === 'org.sagebionetworks.repo.model.FileEntity'
+
+  const { mutate, error } = useLinkFileEntityToURL({
+    onSuccess,
+  })
+
+  useEffect(() => {
+    const isValid = url.trim().length > 0
```

Comment:
Any additional validation needed here?  I'm not sure how much sanitation we do on the rendering side for this URL value throughout the web apps.

Reply:
We can't assume this is set via the web interface, so I think this is fine.

Reply:
This is the same behavior as SWC -- only check that the URL is a nonempty string and let the backend validate syntax